odxtools/parameters/tablekeyparameter.py

```diff
@@ -37,37 +40,107 @@ def _resolve_odxlinks(self, odxlinks: OdxLinkDatabase) -> None:
         super()._resolve_odxlinks(odxlinks)
 
         if self.table_ref:
-            self._table = odxlinks.resolve(self.table_ref)
+            if TYPE_CHECKING:
+                self._table = odxlinks.resolve(self.table_ref, Table)
+            else:
+                self._table = odxlinks.resolve(self.table_ref)
```

Comment:
Why?

Reply:
to prevent mypy complaining about a the type of `._table`. (we can't do this unconditionally because that would lead to a circular import.)

Reply:
You can break the circular import by making the import lazy, and moving it inside the function
Doing import inside function is usually discoureged, but it is one of the simplest and most effective ways to break circular dependency without any side effects
```suggestion
            from .table import Table
            self._table = odxlinks.resolve(self.table_ref, Table)
```

Reply:
yeah, that's used in a few places already. I always wonder what the performance implications of this are, though...

Reply:
the performance implications can be significant:
```python
import timeit
import odxtools

def eager():
    return type(odxtools.load_file)

def lazy():
    import odxtools as odxtools_lazy
    return type(odxtools_lazy.load_file)

t = timeit.timeit("eager()", number=1000*1000, globals=globals())
print(f"eager: {t} seconds elapsed")

t = timeit.timeit("lazy()", number=1000*1000, globals=globals())
print(f"lazy: {t} seconds elapsed")
```

produces
```
eager: 0.038458367984276265 seconds elapsed
lazy: 0.12220152601366863 seconds elapsed
```
on my machine. I'd say performance are acceptable if there is no other solution, but lazy imports should be avoided if possible...